connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp

```diff
@@ -0,0 +1,1720 @@
+
+    if (_radio_events->cad_done)
+    {
+        _radio_events->cad_done((irq_status & IRQ_CAD_ACTIVITY_DETECTED)
+                                == IRQ_CAD_ACTIVITY_DETECTED);
+    }
+}
+
+
+void STM32WL_LoRaRadio::HAL_SUBGHZ_RxTxTimeoutCallback(void)
+{
+    if ((_radio_events->tx_timeout) && (_operation_mode == MODE_TX))
+    {
+        _radio_events->tx_timeout();
+
+        /* Reset TX DBG pin */
+//    DBG_GPIO_RADIO_TX(RST);
+//    STM32WL_LoRaRadio::setTXPin(0);
+
+
+    }
+    else if ((_radio_events && _radio_events->rx_timeout) && (_operation_mode == MODE_RX))
+    {
+        _radio_events->rx_timeout();
+
+        /* Reset RX DBG pin */
+//   DBG_GPIO_RADIO_RX(RST);
+//   STM32WL_LoRaRadio::setRXPin(0);
+    }
+}
+
+/* ----- */
+/* HAL_SUBGHz Callbacks definitions END */
+
+
+/* STM32WL specific BSP Nucleo board functions */
+void STM32WL_LoRaRadio::SUBGRF_SetSwitch(uint8_t paSelect, RFState_t rxtx)
+{
+    RBI_Switch_TypeDef state = RBI_SWITCH_RX;
+
+    if (rxtx == RFSWITCH_TX)
+    {
+        if (paSelect == RFO_LP)
+        {
+            state = RBI_SWITCH_RFO_LP;
+            Radio_SMPS_Set(SMPS_DRIVE_SETTING_MAX);
+        }
+        if (paSelect == RFO_HP)
+        {
+            state = RBI_SWITCH_RFO_HP;
+        }
+    }
+    else
+    {
+        if (rxtx == RFSWITCH_RX)
+        {
+            state = RBI_SWITCH_RX;
+        }
+    }
+    set_antenna_switch(state);
+}
+
+uint8_t STM32WL_LoRaRadio::SUBGRF_SetRfTxPower(int8_t power)
+{
+    uint8_t paSelect = RFO_LP;
+
+    int32_t TxConfig = board_rf_switch_config;
+
+    switch (TxConfig)
+    {
+        case RBI_CONF_RFO_LP_HP:
+        {
+            if (power > 15)
+            {
+                paSelect = RFO_HP;
+            }
+            else
+            {
+                paSelect = RFO_LP;
+            }
+            break;
+        }
+        case RBI_CONF_RFO_LP:
+        {
+            paSelect = RFO_LP;
+            break;
+        }
+        case RBI_CONF_RFO_HP:
+        {
+            paSelect = RFO_HP;
+            break;
+        }
+        default:
+            break;
+    }
+
+    SUBGRF_SetTxParams(paSelect, power, RADIO_RAMP_40_US);
+
+    return paSelect;
+}
+
+
+void STM32WL_LoRaRadio::SUBGRF_SetTxParams(uint8_t paSelect, int8_t power, radio_ramp_time_t rampTime)
+{
+    uint8_t buf[2];
+
+    if (paSelect == RFO_LP)
+    {
+        if (power == 15)
+        {
+            set_pa_config(0x06, 0x00, 0x01, 0x01);
+        }
+        else
+        {
+            set_pa_config(0x04, 0x00, 0x01, 0x01);
+        }
+        if (power >= 14)
+        {
+            power = 14;
+        }
+        else if (power < -17)
+        {
+            power = -17;
+        }
+        write_to_register(REG_OCP, 0x18);   // current max is 80 mA for the whole device
+    }
+    else // rfo_hp
+    {
+        // WORKAROUND - Better Resistance of the SX1262 Tx to Antenna Mismatch, see DS_SX1261-2_V1.2 datasheet chapter 15.2
```

Comment:
workaround ?

Reply:
same comment present on WL BSP: better remove it or let a trace to explain it?